beacon-chain/blockchain/service.go

```diff
@@ -434,7 +434,7 @@ func (s *Service) initializeChainInfo(ctx context.Context) error {
 		if err != nil {
 			return errors.Wrap(err, "could not hash head block")
 		}
-		headState, err := s.beaconDB.HeadState(ctx)
+		headState, err := s.stateGen.StateByRoot(ctx, headRoot)
```

Comment:
If `headState` in DB != `headRoot` generated state, then it is pretty concerning and makes me wonder if there's  missing underlying bug. Can this behavior be reproduced by a test?

Reply:
Not exactly a test but right now in medalla. Our headstate is actually nil, which is why this never works.